walkthroughs/1-on-walkthroughs/walkthrough.adoc

```diff
@@ -0,0 +1,408 @@
+
+You should also add a second paragraph to the preamble which will be displayed when a users clicks on the card. The second paragraph can have more content than the first one.
+
+=== Adding a Preamble to the template
+
+No we're going to work on the cloned walkthrough template.
+
+. Open `walkthroughs/walkthrough.adoc`
+
+. Change the title of the Walkthrough to
+
+----
+My first Walkthtough
+----
+
+. Add an introduction paragraph to your preamble
+
+. Add another paragraph with more information about the walkthrough. Since this is a demo walkthrough you can put anything you like into those paragraphs.
+
+. Commit your changes
++
+----
+git commit -am "add preamble"
+----
+
+. Push your change to the repository
++
+----
+git push upstream master
+----
+
+[type=verification]
+Make sure that all your changes are in the remote repository.
+
+[type=verificationFail]
+Check your git credentials and that you pushed to the right remote.
+
+[time=10]
+== Publishing the Walkthrough in the Webapp
+
+You can add repositories containing one or more Walkthroughs to a Webapp deployed to Openshift.
+
+=== Adding the template repository to the Webapp
+
+We will now add your Walkthrough repository to the Webapp. Take note of the repository URL.
+
+. Open the link:{openshift-host}[Openshift Console]
+
+. Find the *webapp* project and open it
+
+. Find the *tutorial-web-app* deployment and open it
+
+. Switch to the *Environment* tab
+
+. You should see an environment variable with the name *WALKTHROUGH_LOCATIONS*
+
+. Append your reopsitory URL to the value of that variable, separated by a comma, and click *Save*
+
+. This will trigger an automatic redeploy of the Webapp. Once that is finished, refresh the Webapp in your Browser.
+
+. You should now see your own Walkthrough in the dashboard.
+
+[type=verification]
+Make sure that you can see a Walkthrough with the title *My first Walkthrough* in the dasboard.
+
+[type=verificationFail]
+Check that the Webapp has finished redeployment and no errors occurred.
+
+[time=5]
+== Adding tasks
+
+Walkthroughs are organized in tasks. A task becomes one single page in the Webapp. Tasks should separate the contextual blocks of your Walkthrough, e.g. if the user needs to switch to working with a different middleware service then it could become a new task.
+
+Tasks are introduced by a second level heading (`==`).
+
+=== Adding a task to the template
+
+. The template already contains a sample task. Change the name of it to *My first Task*
+
+. The heading of a task should be followed by a short introduction of what the user is about to learn. Add a paragraph with some content.
+
+. Tasks must have a time associated with them. This lets the user know how long the task will approximately take and is also displayed in the Walkthrough overview.
```

Comment:
Might be worthwhile to mention the time is in minutes.

Also, should we let a user know that if they don't add a time it'll be treated as 0 minutes.

Reply:
I think we can add that as a requirement because we are adding a CI check that enforces a time of >0 on a task.